router/middleware/session/repo.go

```diff
@@ -157,11 +157,13 @@ func MustPull(c *gin.Context) {
 
 	// debugging
 	if user != nil {
-		c.AbortWithStatus(http.StatusNotFound)
+		c.String(http.StatusNotFound, http.StatusText(http.StatusNotFound))
```

Comment:
I don't think it's very informative either, but figured the 404 was returned to not reveal information about a private repository if it actually was missing read permissions instead of a true 404.

I can change it to reflect the same verbosity and content as the debug message below if you'd like.